core/server/master/src/main/java/alluxio/master/file/AccessTimeUpdater.java

```diff
@@ -0,0 +1,176 @@
+
+  /** Keep track of all inodes that need access time update. */
+  private ConcurrentHashSet<Long> mAccessTimeUpdates;
+  private ScheduledExecutorService mExecutorService = null;
+  private AtomicBoolean mUpdateScheduled = new AtomicBoolean();
+
+  /**
+   * Constructs a new {@link AccessTimeUpdater}.
+   */
+  public AccessTimeUpdater(FileSystemMaster fileSystemMaster, InodeTree inodeTree,
+      JournalSystem journalSystem) {
+    this(fileSystemMaster, inodeTree, journalSystem,
+        ServerConfiguration.getMs(PropertyKey.MASTER_FILE_ACCESS_TIME_JOURNAL_FLUSH_INTERVAL),
+        ServerConfiguration.getMs(PropertyKey.MASTER_FILE_ACCESS_TIME_UPDATE_PRECISION),
+        ServerConfiguration.getMs(PropertyKey.MASTER_FILE_ACCESS_TIME_UPDATER_SHUTDOWN_TIMEOUT));
+  }
+
+  /**
+   * Constructs a new {@link AccessTimeUpdater} with time configurations.
+   */
+  @VisibleForTesting
+  public AccessTimeUpdater(FileSystemMaster fileSystemMaster, InodeTree inodeTree,
+      JournalSystem journalSystem, long flushInterval, long updatePrecision, long shutdownTimeout) {
+    mFileSystemMaster = fileSystemMaster;
+    mInodeTree = inodeTree;
+    mAccessTimeUpdates = new ConcurrentHashSet<>();
+    mFlushInterval = flushInterval;
+    mUpdatePrecision = updatePrecision;
+    mShutdownTimeout = shutdownTimeout;
+    journalSystem.addJournalSink(mFileSystemMaster, this);
+  }
+
+  public void start() {
+    start(mFlushInterval > 0 ? Executors.newSingleThreadScheduledExecutor(
+        ThreadFactoryUtils.build("AccessTimeUpdater-%d", true)) : null);
+  }
+
+  @VisibleForTesting
+  public void start(ScheduledExecutorService executorService) {
+    mExecutorService = executorService;
+  }
+
+  @Override
+  public void beforeShutdown() {
+    if (mExecutorService != null) {
+      flushUpdates();
+    }
+  }
+
+  public void stop() {
+    if (mExecutorService != null) {
+      ThreadUtils.shutdownAndAwaitTermination(mExecutorService, mShutdownTimeout);
+    }
+  }
+
+  /**
+   * Update inode last access time.
+   *
+   * @param context the journal context
+   * @param inode the target inode
+   * @param opTimeMs the last access time to set on the inode
+   */
+  public void updateAccessTime(JournalContext context, Inode inode, long opTimeMs) {
+    if (opTimeMs - inode.getLastAccessTimeMs() > mUpdatePrecision) {
+      try (LockResource lr = mInodeTree.getInodeLockManager().lockUpdate(inode.getId())) {
+        UpdateInodeEntry entry = mInodeTree.updateInodeAccessTime(inode.getId(), opTimeMs);
+        if (mExecutorService != null) {
+          scheduleJournalUpdate(entry);
+        } else {
+          journalAccessTime(context, entry);
+        }
+      }
+    }
+  }
+
+  private void scheduleJournalUpdate(UpdateInodeEntry entry) {
+    mAccessTimeUpdates.add(entry.getId());
+    if (mUpdateScheduled.compareAndSet(false, true)) {
+      mExecutorService.schedule(this::flushScheduledUpdates, mFlushInterval, TimeUnit.MILLISECONDS);
```

Comment:
can you use this instead? https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html#scheduleWithFixedDelay(java.lang.Runnable,%20long,%20long,%20java.util.concurrent.TimeUnit)